src/main/java/org/cbioportal/web/GenePanelDataController.java

```diff
@@ -83,9 +84,10 @@ public ResponseEntity<List<GenePanelData>> fetchGenePanelDataInMultipleMolecular
         @Parameter(required = true, description = "Gene panel data filter object")
         @RequestBody(required = false) GenePanelDataMultipleStudyFilter genePanelDataMultipleStudyFilter) {
 
-        List<GenePanelData> genePanelDataList;
-        if(CollectionUtils.isEmpty(interceptedGenePanelDataMultipleStudyFilter.getMolecularProfileIds())) {
-            List<MolecularProfileCaseIdentifier> molecularProfileSampleIdentifiers = interceptedGenePanelDataMultipleStudyFilter.getSampleMolecularIdentifiers()
+        List<GenePanelData> genePanelDataList = new ArrayList<>();
+        if(interceptedGenePanelDataMultipleStudyFilter.getMolecularProfileIds()!= null && CollectionUtils.isEmpty(interceptedGenePanelDataMultipleStudyFilter.getMolecularProfileIds())) {
```

Comment:
it would be nicer to call the endpoint once, store it to a variable and test against it - a function call might be unnecessarily expensive.